src/Compilers/Core/Portable/Compilation/CompilationOptions.cs

```diff
@@ -226,6 +227,11 @@ public abstract class CompilationOptions
         public StrongNameProvider StrongNameProvider { get; protected set; }
 
         /// <summary>
+        /// Provides options for analyzer consumption that apply to specific syntax trees. Null if no syntax tree specific options exist.
```

Comment:
I agree that the "non-null" pattern is better, but chose to be consistent with the other Resolvers/Providers in this type. If I had a time machine though...